src/Users/Utente.java

```diff
@@ -78,5 +81,13 @@ public Roles getRole() {
         return role;
     }
 
+    public Carrello getCarrelloCliente() {
 
+        if(carrelloCliente == null) carrelloCliente = new Carrello();
```

Comment:
Come mai questo controllo? Se carrelloCliente è null vuol dire che qualcuno ha fatto setCarrelloCliente(null) e così voleva che fosse, per cui sarebbe giusto che il getter ritornasse null.

Se non è possibile che il carrello sia null, il controllo va aggiunto nel setter ed eventualmente va lanciata un'eccezione.

In generale i getter non dovrebbero contenere logica, dovrebbero ritornare i valori per come sono. Se ci sono dei vincoli da aggiungere ai dati, essi vanno controllati quando vengono modificati i valori (costruttori e setter).

Reply:
Il valore null mi veniva restituito nel caso ne Json ci fosse un carrello vuoto